spark-bigquery-dsv2/spark-3.4-bigquery/src/test/java/com/google/cloud/spark/bigquery/integration/Spark34IndirectWriteIntegrationTest.java

```diff
@@ -31,5 +50,47 @@ public void setParquetLoadBehaviour() {
   }
 
   // tests from superclass
+  @Test
+  public void testTimestampNTZWriteToBigQueryAvroFormat() throws InterruptedException {
+    LocalDateTime time = LocalDateTime.of(2023, 9, 1, 12, 23, 34, 268543 * 1000);
+    List<Row> rows = Arrays.asList(RowFactory.create(time));
+    Dataset<Row> df =
+        spark.createDataFrame(
+            rows,
+            new StructType(
+                new StructField[] {
+                  StructField.apply("foo", DataTypes.TimestampNTZType, true, Metadata.empty())
+                }));
+    writeToBigQuery(df, SaveMode.Overwrite, "avro");
+    BigQuery bigQuery = IntegrationTestUtils.getBigquery();
+    TableResult result =
+        bigQuery.query(
+            QueryJobConfiguration.of(String.format("Select foo from %s", fullTableName())));
+    assertThat(result.getSchema().getFields().get(0).getType())
+        .isEqualTo(LegacySQLTypeName.DATETIME);
+    assertThat(result.streamValues().findFirst().get().get(0).getValue().toString())
+        .isEqualTo(time.format(DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSSSS")));
```

Comment:
Please replace with actual string representation instead of time.format(...)

Reply:
Fixed